platform/src/main/python/dlpx/virtualization/platform/_linked.py

```diff
@@ -141,10 +142,14 @@ def _internal_direct_pre_snapshot(self, request):
             json.loads(request.repository.parameters.json))
         source_config = SourceConfigDefinition.from_dict(
             json.loads(request.source_config.parameters.json))
+        snapshot_parameters = SnapshotParametersDefinition.from_dict(
+            json.loads(request.snapshot_parameters.parameters.json))
 
-        self.pre_snapshot_impl(direct_source=direct_source,
-                               repository=repository,
-                               source_config=source_config)
+        self.pre_snapshot_impl(
+            direct_source=direct_source,
+            repository=repository,
+            source_config=source_config,
+            optional_snapshot_parameters=snapshot_parameters)
```

Comment:
I love the idea of putting the word "optional" right in the argument name here.

Reply:
haha it was your idea! so thank you 🥇

Reply:
It was?? I have no memory of this.